contracts/EthlanceJobs.sol

```diff
@@ -0,0 +1,851 @@
+
+  address public owner; // The address of the individual who's allowed to set the metaTxRelayer address
+  address public metaTxRelayer; // The address of the meta transaction relayer whose _sender is automatically trusted for all contract calls
+
+  bool public callStarted; // Ensures mutex for the entire contract
+
+  /*
+   * Modifiers
+   */
+
+  modifier callNotStarted(){
+    require(!callStarted);
+    callStarted = true;
+    _;
+    callStarted = false;
+  }
+
+  modifier validateJobArrayIndex(
+                                 uint _index)
+  {
+    require(_index < numJobs);
+    _;
+  }
+
+  modifier validateContributionArrayIndex(
+                                          uint _jobId,
+                                          uint _index)
+  {
+    require(_index < jobs[_jobId].contributions.length);
+    _;
+  }
+
+  modifier validateFulfillmentArrayIndex(
+                                         uint _jobId,
+                                         uint _index)
+  {
+    require(_index < jobs[_jobId].fulfillments.length);
+    _;
+  }
+
+  modifier validateIssuerArrayIndex(
+                                    uint _jobId,
+                                    uint _index)
+  {
+    require(_index < jobs[_jobId].issuers.length);
+    _;
+  }
+
+  modifier validateApproverArrayIndex(
+                                      uint _jobId,
+                                      uint _index)
+  {
+    require(_index < jobs[_jobId].approvers.length);
+    _;
+  }
+
+  modifier onlyIssuer(
+                      address _sender,
+                      uint _jobId,
+                      uint _issuerId)
+  {
+    require(_sender == jobs[_jobId].issuers[_issuerId]);
+    _;
+  }
+
+  modifier onlySubmitter(
+                         address _sender,
+                         uint _jobId,
+                         uint _fulfillmentId)
+  {
+    require(_sender ==
+            jobs[_jobId].fulfillments[_fulfillmentId].submitter);
+    _;
+  }
+
+  modifier onlyContributor(
+                           address _sender,
+                           uint _jobId,
+                           uint _contributionId)
+  {
+    require(_sender ==
+            jobs[_jobId].contributions[_contributionId].contributor);
+    _;
+  }
+
+  modifier isApprover(
+                      address _sender,
+                      uint _jobId,
+                      uint _approverId)
+  {
+    require(_sender == jobs[_jobId].approvers[_approverId]);
+    _;
+  }
+
+  modifier hasNotPaid(
+                      uint _jobId)
+  {
+    require(!jobs[_jobId].hasPaidOut);
+    _;
+  }
+
+  modifier hasNotRefunded(
+                          uint _jobId,
+                          uint _contributionId)
+  {
+    require(!jobs[_jobId].contributions[_contributionId].refunded);
+    _;
+  }
+
+  modifier senderIsValid(
+                         address _sender)
+  {
+    require(msg.sender == _sender || msg.sender == metaTxRelayer);
+    _;
+  }
```

Comment:
I don't think these needs to be modifiers, private functions.

Reply:
I think the same but I think we should leave them like that for consistency with StandardBouties, so we can compare both contracts which I find is important